specification/appconfiguration/resource-manager/Microsoft.AppConfiguration/preview/2023-08-01-preview/appconfiguration.json

```diff
@@ -0,0 +1,2890 @@
+          "enum": [
+            "Creating",
+            "Updating",
+            "Deleting",
+            "Succeeded",
+            "Failed",
+            "Canceled"
+          ],
+          "type": "string",
+          "readOnly": true,
+          "x-ms-enum": {
+            "name": "ProvisioningState",
+            "modelAsString": true
+          }
+        },
+        "status": {
+          "description": "The current status of the snapshot.",
+          "enum": [
+            "Provisioning",
+            "Ready",
+            "Archived",
+            "Failed"
+          ],
+          "type": "string",
+          "readOnly": true,
+          "x-ms-enum": {
+            "name": "SnapshotStatus",
+            "modelAsString": true
+          }
+        },
+        "filters": {
+          "description": "A list of filters used to filter the key-values included in the snapshot.",
+          "type": "array",
+          "minItems": 1,
+          "maxItems": 3,
+          "items": {
+            "$ref": "#/definitions/KeyValueFilter"
+          },
+          "x-ms-identifiers": []
+        },
+        "compositionType": {
+          "description": "The composition type describes how the key-values within the snapshot are composed. The 'key' composition type ensures there are no two key-values containing the same key. The 'key_label' composition type ensures there are no two key-values containing the same key and label.",
+          "type": "string",
+          "enum": [
+            "Key",
+            "Key_Label"
+          ],
+          "x-ms-enum": {
+            "name": "CompositionType",
+            "modelAsString": true
+          }
+        },
+        "created": {
+          "description": "The time that the snapshot was created.",
+          "format": "date-time",
+          "type": "string",
+          "readOnly": true
+        },
+        "expires": {
+          "description": "The time that the snapshot will expire.",
+          "format": "date-time",
+          "type": "string",
+          "readOnly": true
+        },
+        "retentionPeriod": {
+          "description": "The amount of time, in seconds, that a snapshot will remain in the archived state before expiring. This property is only writable during the creation of a snapshot. If not specified, the default lifetime of key-value revisions will be used.",
+          "type": "integer",
+          "minimum": 3600,
+          "maximum": 7776000,
+          "format": "int64",
+          "x-ms-mutability": [
+            "create",
+            "read"
+          ]
+        },
+        "size": {
+          "description": "The size in bytes of the snapshot.",
+          "type": "integer",
+          "format": "int64",
+          "readOnly": true
+        },
+        "itemsCount": {
+          "description": "The amount of key-values in the snapshot.",
+          "type": "integer",
+          "format": "int64",
+          "readOnly": true
+        },
+        "tags": {
```

Comment:
>tags [](http://example.com/codeflow?start=9&length=4)

Is this a tracked resource? If so, you should use the TrackedResource definition from common types. If it's not a tracked resource type, don't create a property named tags, since that conflicts with ARM tags. If it's a proxy type, but you want tag functionality, you should consider making it a tracked resource type.

Reply:
No, this is a proxy for a data plane snapshot which has its own tags (not ARM tags). This is modeled similarly to the existing KeyValue resource, which is a proxy for a data plane KeyValue with its own tags too:

https://github.com/Azure/azure-rest-api-specs/blob/main/specification/appconfiguration/resource-manager/Microsoft.AppConfiguration/stable/2023-03-01/appconfiguration.json#L1819C10-L1819C14

Could you elaborate on the conflict with ARM tags?

Reply:
Got it. The part that it is not ARM Tags clarifies and also, it is inside properties bag, so that is good. Can this be named something not tags, to avoid confusing users with the tags name?

Reply:
Because this is a proxy for a data plane snapshot, we would like to keep the same properties as in the data plane API (which uses the name tags).

https://github.com/Azure/azure-rest-api-specs/blob/main/specification/appconfiguration/data-plane/Microsoft.AppConfiguration/stable/2023-10-01/appconfiguration.json#L1714C10-L1714C14

Note: I updated the property description to clarify that these are not ARM tags.